### PR TITLE
Change nil to null in doc

### PR DESCRIPTION
**Pull request by einverne**
ref #2332

### CODE REVIEW

File: src/main/java/redis/clients/jedis/BinaryJedis.java [hunk 1]
```diff
@@ -857,7 +857,7 @@ public String flushAll(FlushMode flushMode) {
 
   /**
    * Get the values of all the specified keys. If one or more keys don't exist or is not of type
-   * String, a 'nil' value is returned instead of the value of the specified key, but the operation
+   * String, a 'null value is returned instead of the value of the specified key, but the operation
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/BinaryJedis.java:860]: I think both are fine. According to the [redis doc](https://redis.io/commands/mget). It should be `nil`. In java, `null` is also make senses. BTW, `'null` should be `'null'`

**Review by dengliming — CHANGES_REQUESTED** [review]

**Comment by sazzad16** [comment]
@dengliming I object to this change. FYI https://github.com/redis/jedis/pull/2268#issuecomment-766351904

**Comment by einverne** [comment]
> @dengliming I object to this change. FYI [#2268 (comment)](https://github.com/redis/jedis/pull/2268#issuecomment-766351904)

Oh, sorry, I misunderstand your intention in #2332, I believe we just need one place to explain this clearly. It's not necessary to update the doc everywhere in the codes. So I will close the PR.

**Comment by sazzad16** [comment]
@einverne There is nothing to be sorry about. And you did not misunderstand. During #2332, I was not against using `null`. But as I was thinking about #2332 and #2268, I was preferring `nil` more and more.

**Comment by devxzero** [comment]
'nil' is just plain wrong from a Java perspective. And since this is Java language, it it should be documented as null.
A lot of people who are experienced with Java, but new to Redis, will get a nullpointerexception, and wonder: _why is it returning null, the doc says "the special value 'nil' is returned"_ The 'nil' is even between quotes, which suggests that it should be taken literary as a literal string. Yes, I did wonder: why didn't they return just null? I thought, maybe Jedis wants to stick as low level as possible, so it doesn't want to translate 'nil' to null.
It causes people who are new to Redis to have to debug the Jedis code to understand why this problem exists. After some googling, I found multiple issues about this problem on this repo.

**Comment by sazzad16** [comment]
@devxzero or anyone else interested, would you like to craft a PR? The PR should cover all 'nil' (as much as possible) and use more modern javadoc styles (e.g. `{@code null}`).

**Pull request closed without merging**